Change recorder to my fork, fix breakage caused by recent Decky changes

safijari opened the pull request:
# Decky Recorder
Records games.

I'm requesting that we change the plugin over to my fork which is currently maintained. It contains fixes for some common bugs as well as a recent breakage caused by 2.10.4 (see discussion here https://github.com/marissa999/decky-recorder/issues/65).

## Checklist:

### Developer Checklist

- [X] I am the original author or an authorized maintainer of this plugin.
- [X] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [X] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [X] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

<!-- The following section needs to be modified as yes/no answers by the plugin developer. -->

<!-- Ex: "**Yes/No**: ..." becomes "**Yes**: ..." -->

### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is should be modified to fit the conditions for plugin testing found here: https://wiki.deckbrew.xyz/en/plugin-dev/review-and-testing -->

## Testing

<!-- Remove this box for SteamOS Stable/Beta testing if you use a custom or remote binary, for more info follow the URL in the comment above the testing section. -->
- [ ] Tested on SteamOS Stable/Beta Update Channel.

TrainDoctor commented:
Thanks for your submission. Just want to make sure @marissa999 gave this the go ahead. I've reviewed the code changes you made and they look good so far, I just want to make sure to not step on any toes. Thanks.

safijari commented:
Sounds good. They have not been seen in some time (likely got very busy with school) and currently my main interest is in putting out a working build of the plugin and I can't quite do that in the main repo without making more changes there without the involvement of the original author.

TrainDoctor commented:
> Sounds good. They have not been seen in some time (likely got very busy with school) and currently my main interest is in putting out a working build of the plugin and I can't quite do that in the main repo without making more changes there without the involvement of the original author.

Makes sense. My main concern being keeping the original author in the loop as needed but also making sure folks know that this is effectively an independent but trustworthy fork etc.

safijari commented:
> > Sounds good. They have not been seen in some time (likely got very busy with school) and currently my main interest is in putting out a working build of the plugin and I can't quite do that in the main repo without making more changes there without the involvement of the original author.
> 
> Makes sense. My main concern being keeping the original author in the loop as needed but also making sure folks know that this is effectively an independent but trustworthy fork etc.

Would like to understand next steps. Is the intro to wait some amount of time (how long?) to allow the original author to respond? Or would you want to wait indefinitely until they respond? 

What are some alternative ways for me to get the fix in people's hands? There's been a lot of comments indicating the breakage in the last few days and people are resorting to downgrading Decky. I can always release is as a zip with install instructions but would like to avoid that. I could rename it to explicitly indicate that it's not the same plugin but that would lead to fragmentation.

Thoughts?

TrainDoctor commented:
> Would like to understand next steps. Is the intent to wait some amount of time (how long?) to allow the original author to respond? Or would you want to wait indefinitely until they respond?
> 
> What are some alternative ways for me to get the fix in people's hands? There's been a lot of comments indicating the breakage in the last few days and people are resorting to downgrading Decky. I can always release it as a zip with install instructions but would like to avoid that. I could rename it to explicitly indicate that it's not the same plugin but that would lead to fragmentation.
> 
> Thoughts?

I don't have an easy answer for this to be honest right now. I'll try to get back to you when I can. For now I will send this to the testing store so users can access it for now.

nabel0 reviewed (changes requested):
Tested an wont work on my SteamDeck (StableOS).
After Switching "Replay Mode" to "OFF" it allways switches to ON by itselfe if i leave the Plugin UI.
It also wont record anything.
[decky-recorder.log](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12438093/decky-recorder.log)

safijari commented:
> Tested an wont work on my SteamDeck (StableOS). After Switching "Replay Mode" to "OFF" it allways switches to ON by itselfe if i leave the Plugin UI. It also wont record anything. [decky-recorder.log](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12438093/decky-recorder.log)

Yeah it looks like there's a difference between how it builds locally for me vs how it builds in this repo. Locally for me building the "backend" (it's just vendoring some gstreamer stuff and psutil) puts some files in `backend/out` which I later reference in the python code. In this repo those files end up under `bin`.

How should I deal with that difference?

nabel0 reviewed (changes requested):
Sadly this is still not working
[decky-recorder.log](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12443785/decky-recorder.log)
[decky-recorder-std-err.log](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12443786/decky-recorder-std-err.log)

safijari commented:
Yeah I made a mistake. Continued developing after I had opened the original PR and then did my fix for the discrepancy between build systems on top of an untested change. I'll update this and make sure it's fully tested this time.

safijari commented:
Third time's the charm

nabel0 reviewed (approved):
Works fine now.  Tested on SteamOS Stable/Beta Update

safijari commented:
A number of other testers have also indicated that it's working for them now.

grimkor commented:
Having some troubles with the latest version of this, it either records a 0 byte video or replay mode will say "too early to record another clip"

![decky-recorder-std-out.log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448847/decky-recorder-std-out.log.2.txt)

![decky-recorder (2).log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448845/decky-recorder.2.log.2.txt)

![decky-recorder-std-err.log (3).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448846/decky-recorder-std-err.log.3.txt)

Attached logs.

safijari commented:
> Having some troubles with the latest version of this, it either records a 0 byte video or replay mode will say "too early to record another clip"
> 
> ![decky-recorder-std-out.log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448847/decky-recorder-std-out.log.2.txt)
> 
> ![decky-recorder (2).log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448845/decky-recorder.2.log.2.txt)
> 
> ![decky-recorder-std-err.log (3).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448846/decky-recorder-std-err.log.3.txt)
> 
> Attached logs.

Please try uninstalling and then installing the latest from the testing store.

The errors imply that maybe some old files are still left

grimkor commented:
> > Having some troubles with the latest version of this, it either records a 0 byte video or replay mode will say "too early to record another clip"
> > 
> > ![decky-recorder-std-out.log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448847/decky-recorder-std-out.log.2.txt)
> > 
> > ![decky-recorder (2).log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448845/decky-recorder.2.log.2.txt)
> > 
> > ![decky-recorder-std-err.log (3).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448846/decky-recorder-std-err.log.3.txt)
> > 
> > Attached logs.
> 
> Please try uninstalling and then installing the latest from the testing store.
> 
> The errors imply that maybe some old files are still left

Hey did this and same problem, dropped down to Stable and it worked. Moved back to Main and same errors.

safijari commented:
> > > Having some troubles with the latest version of this, it either records a 0 byte video or replay mode will say "too early to record another clip"
> > > ![decky-recorder-std-out.log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448847/decky-recorder-std-out.log.2.txt)
> > > ![decky-recorder (2).log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448845/decky-recorder.2.log.2.txt)
> > > ![decky-recorder-std-err.log (3).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448846/decky-recorder-std-err.log.3.txt)
> > > Attached logs.
> > 
> > 
> > Please try uninstalling and then installing the latest from the testing store.
> > The errors imply that maybe some old files are still left
> 
> Hey did this and same problem, dropped down to Stable and it worked. Moved back to Main and same errors.

Oh. Main.

Yeah this doesn't work in main. There's an incompatibility between new versions of pipewire and gamescope. I've had an issue opened in the gamescope repo for a long time which has gotten no traction.

It also seems to affect in home streaming so hopefully they'll fix it before 3.5 comes out. Otherwise Recorder will be dead in the water.

grimkor commented:
> > > > Having some troubles with the latest version of this, it either records a 0 byte video or replay mode will say "too early to record another clip"
> > > > ![decky-recorder-std-out.log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448847/decky-recorder-std-out.log.2.txt)
> > > > ![decky-recorder (2).log (2).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448845/decky-recorder.2.log.2.txt)
> > > > ![decky-recorder-std-err.log (3).txt](https://github.com/SteamDeckHomebrew/decky-plugin-database/files/12448846/decky-recorder-std-err.log.3.txt)
> > > > Attached logs.
> > > 
> > > 
> > > Please try uninstalling and then installing the latest from the testing store.
> > > The errors imply that maybe some old files are still left
> > 
> > Hey did this and same problem, dropped down to Stable and it worked. Moved back to Main and same errors.
> 
> Oh. Main.
> 
> Yeah this doesn't work in main. There's an incompatibility between new versions of pipewire and gamescope. I've had an issue opened in the gamescope repo for a long time which has gotten no traction.
> 
> It also seems to affect in home streaming so hopefully they'll fix it before 3.5 comes out. Otherwise Recorder will be dead in the water.

I see my mistake, sorry about that.

safijari commented:
Not your mistake at all. Thank you for testing :)

safijari commented:
@TrainDoctor thoughts on next steps?

TrainDoctor commented:
> @TrainDoctor thoughts on next steps?

I'll likely merge tomorrow (time permitting) due to original author's lack of response.
Note to self: We should really implement an MOTD for stuff like this...

TrainDoctor commented:
In future there will be a notification sent out to users of a plugin if the maintainer changes. Thanks for your contributions @safijari!

MediaWithMiles commented:
So what do us users do now, just wait? Which I am totally fine with.

TrainDoctor commented:
> So what do us users do now, just wait? Which I am totally fine with.

The update is out. Feel free to install the latest version if you're on stable/beta steam client.
